modules/extras.py

```diff
@@ -226,6 +232,7 @@ def filename_nothing():
 
     filename = filename_generator() if custom_name == '' else custom_name
     filename += ".inpainting" if result_is_inpainting_model else ""
+    filename += ".pix2pix" if result_is_pix2pix_model else ""

```

Comment:
My suggestion to use the word "instruct" in the two places above might, or might not, also apply here. Did you invent this file format name, or is this already used elsewhere? If you're inventing it, I might suggest something like `.instrpix2pix` or `.ip2p`. Disregard if you think this comment is not helpful.

Reply:
@Klace what do you think about naming it `.ip2p` rather than `.pix2pix`? My understanding is that the academic name for img2img is actually pix2pix but somehow it got named img2img in mainstream usage, so this "instruct" approach is just using the existing academic name "pix2pix" to mean the same thing as "img2img". So the "instruct" part is the novel aspect, not the "pix2pix" part. I'd therefore say it is rather confusing to call it `pix2pix`. Is it too late to change the file extension? I'm curious if you agree or not with this reasoning and consider it worth a change.

Reply:
It looks like Klace's extension simply checks for "pix2pix" to be in the filename so I tentatively changed the extension to be ".instrpix2pix".  This way we can make the change and it still works with the extension as is.

If you guys decide to change the extension name further I have no problem changing my code to conform with whatever you decide.

Reply:
My recommendation is to keep the extension as .ckpt or .safetensors, as that indicates the overall file format, and only append instruct-pix2pix to the base filename.

Reply:
Sorry I might have misspoke calling it an extension.  It actually still has the same endings on the filename as before.  It just has instrpix2pix right before the extension now.

When the merge tab generates inpainting models it's always something like MyModel.inpainting.ckpt or MyModel.inpainting.safetensors.

So I just made pix2pix work the same way, with MyModel.instrpix2pix.ckpt or MyModel.instrpix2pix.safetensors.

Reply:
> @Klace what do you think about naming it `.ip2p` rather than `.pix2pix`? My understanding is that the academic name for img2img is actually pix2pix but somehow it got named img2img in mainstream usage, so this "instruct" approach is just using the existing academic name "pix2pix" to mean the same thing as "img2img". So the "instruct" part is the novel aspect, not the "pix2pix" part. I'd therefore say it is rather confusing to call it `pix2pix`. Is it too late to change the file extension? I'm curious if you agree or not with this reasoning and consider it worth a change.

I am concerned that "instruct" as the keyword could cause issues, for instance if someone creates a model for "Instructional_Material.ckpt" or "Sexy Instructors.ckpt" just as some theoretical model names that someone . I agree using the special token to hijack isn't the best solution due to this. I was not aware of the original academic name for img2img being pix2pix (makes sense).

If we use a new unique name then users will have to rename the model when downloading from the creator. I suppose we can use the full "instruct-pix2pix" keyword instead, that would safest to me and supports the original model filename.